<launch>
	<!-- global rosparam -->
	<param name="use_sim_time" value="true"/>

	<!-- rosbag -->
	<node pkg="rosbag" type="play" name="player" args="--clock /home/amsl/ozaki/bagfiles/infant/mocap/0113_infant/2019-01-13-15-46-58.bag -s 60"/>
	<!-- <node pkg="rosbag" type="play" name="player" args="&#45;&#45;clock /home/amsl/ozaki/bagfiles/infant/mocap/0113_infant/2019&#45;01&#45;13&#45;15&#45;46&#45;58.bag &#45;s 130"/> -->

	<!-- semi sensor driver -->
	<node pkg="image_transport" type="republish" name="republish" args="compressed raw">
		<remap from="in" to="/camera/color/image_raw"/>
		<remap from="out" to="/camera/color/image_raw/decompressed"/>
	</node>
	<node pkg="msg_conversion" name="infant_odometry_republish" type="infant_odometry_republish" output="screen">
		<!-- <param name="velocity_scalar_ratio" type="double" value="0.92"/> -->
	</node>

	<!-- rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find save_dataset)/rviz_config/save_image_with_imu.rviz"/>

	<!-- main -->
	<node pkg="save_dataset" name="save_image_with_imu" type="save_image_with_imu" output="screen">
		<remap from="/image_raw" to="/camera/color/image_raw/decompressed"/>
		<remap from="/odom" to="/tinypower/odom/republished"/>
		<param name="save_dir_path" type="string" value="$(find save_dataset)/dataset"/>
		<!-- <param name="save_csv_name" type="string" value="$(find save_dataset)/dataset/save_image_with_imu.csv"/> -->
		<param name="save_data_limit" type="int" value="50"/>
		<param name="th_diff_position" type="double" value="10.0"/>
	</node>
</launch>
